pkg/agent/baker.go

```diff
@@ -826,10 +826,23 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
 			}
 			return 0
 		},
+		"ShouldConfigureCustomCATrust": func() bool {
+			return areCustomCATrustCertsPopulated(*config)
+		},
+		"GetCustomCATrustConfigCerts": func() []string {
+			if areCustomCATrustCertsPopulated(*config) {
+				return config.CustomCATrustConfig.CustomCATrustCerts
+			}
+			return []string{}
+		},
 	}
 }
 
 func isStandardNCv1(size string) bool {
 	tmp := strings.ToLower(size)
 	return strings.HasPrefix(tmp, "standard_nc") && !strings.Contains(tmp, "_v")
 }
+
+func areCustomCATrustCertsPopulated(config datamodel.NodeBootstrappingConfiguration) bool {
+	return config.CustomCATrustConfig != nil && config.CustomCATrustConfig.CustomCATrustCerts != nil
```

Comment:
nit
```suggestion
	return config.CustomCATrustConfig != nil && len(config.CustomCATrustConfig.CustomCATrustCerts) > 0
```

https://go.dev/play/p/y_7HlM92Phr

Reply:
don't know if we validate the empty array on rp side or not

Reply:
lol I see cameron left the opposite review, but I think keeping len > 0 and skipping the nil check is correct?

Reply:
On rp side empty array will be treated as a valid input, but on AB side we don't want to do anything with that. I think both of these will work. Discussed on the side with Cameron that switching from len to nil check there might be less readable so I'm okay with letting it stay like it was originally.

Reply:
yep, either works fine, I was mainly just thinking we should be consistent in what we choose there :)